diff -up gnunet-0.19.4/configure.ac.1~ gnunet-0.19.4/configure.ac
--- gnunet-0.19.4/configure.ac.1~	2023-02-06 06:51:10.000000000 +0100
+++ gnunet-0.19.4/configure.ac	2023-09-13 15:30:23.684810804 +0200
@@ -630,10 +630,10 @@ AC_DEFUN([CHECK_WITH_LIB],
          [$1], [$2], [$3],
          [eval "AS_TR_SH([$1])=1"],
          [eval "AS_TR_SH([$1])=0"],
-         [-L${withval}/lib],
-         [-I${withval}/include])
+         [],
+         [])
        AS_IF([test "x$][AS_TR_SH([$1])][" = "x1"],
-         [EXT_LIB_PATH="-L${withval}/lib $EXT_LIB_PATH"])])],
+         [EXT_LIB_PATH="$EXT_LIB_PATH"])])],
     [CHECK_LIBHEADER([m4_translit([$1], [a-z], [A-Z])],
          [$1], [$2], [$3],
          [eval "AS_TR_SH([$1])=1"],
@@ -805,8 +805,8 @@ AS_IF([test "x$libidn" = "xyes"],
      [AC_CHECK_HEADERS([idn2/idn2.h],
                        [libidn_version="2"],
                        [libidn_version="0"])])],
-  [LDFLAGS="-L$libidn/lib $LDFLAGS"
-   CPPFLAGS="-I$libidn/include $CPPFLAGS"
+  [LDFLAGS="$LDFLAGS"
+   CPPFLAGS="$CPPFLAGS"
    AC_CHECK_HEADERS([idn2.h],
      [libidn_version="2"],
      [AC_CHECK_HEADERS([idn2/idn2.h],
@@ -820,8 +820,8 @@ AS_IF([test "x$libidn_version" = "x0"],
         [AC_CHECK_HEADERS([idn/idna.h],
                           [libidn_version="1"]
                           [libidn_version="2"])])],
-     [LDFLAGS="-L$libidn/lib $LDFLAGS"
-      CPPFLAGS="-I$libidn/include $CPPFLAGS"
+     [LDFLAGS="$LDFLAGS"
+      CPPFLAGS="$CPPFLAGS"
       AC_CHECK_HEADERS([idna.h],
         [libidn_version="2"],
         [AC_CHECK_HEADERS([idn/idna.h],
@@ -856,8 +856,8 @@ AS_IF([test "x$zlib" = "xyes"],
   [AC_CHECK_HEADER([zlib.h],
      [AC_CHECK_LIB([z], [compress2], [zlib_ok=1], [zlib_ok=0])],
      [zlib_ok=0])],
-  [LDFLAGS="-L$zlib/lib $LDFLAGS"
-   CPPFLAGS="-I$zlib/include $CPPFLAGS"
+  [LDFLAGS="$LDFLAGS"
+   CPPFLAGS="$CPPFLAGS"
    AC_CHECK_HEADER([zlib.h],
      [AC_CHECK_LIB([z], [compress2], [zlib_ok=1], [zlib_ok=0])],
      [zlib_ok=0])])
@@ -866,8 +866,8 @@ AS_IF([test "x$zlib_ok" = "x0"], [AC_MSG
 AS_IF([test "x$zlib" = "xyes"],
   [Z_CFLAGS=""
    Z_LIBS="-lz"],
-  [Z_CFLAGS="-I${zlib}/include"
-   Z_LIBS="-L${zlib}/lib -lz"])
+  [Z_CFLAGS=""
+   Z_LIBS="-lz"])
 AC_SUBST([Z_CFLAGS])
 AC_SUBST([Z_LIBS])
 
@@ -912,24 +912,20 @@ AC_ARG_WITH([mysql],
      [AC_CHECK_HEADERS([mysql/mysql.h],
         [AC_CHECK_LIB([mysqlclient], [mysql_init],
                       [mysql=true], [mysql=false])])],
-     [LDFLAGS="-L$with_mysql/lib -L$with_mysql/lib/mysql $LDFLAGS $ZLIBS"
-      CPPFLAGS="-I$with_mysql/include $CPPFLAGS"
+     [LDFLAGS="$LDFLAGS $ZLIBS"
       AC_CHECK_HEADERS([mysql/mysql.h],
         [AC_CHECK_LIB([mysqlclient], [mysql_init],
-                      [MYSQL_LDFLAGS="-L$with_mysql/lib -L$with_mysql/lib/mysql"
-                       MYSQL_CPPFLAGS="-I$with_mysql/include"
-                       mysql=true],
+                      [mysql=true],
                       [mysql=false])])])],
    [AS_IF([test -d "/usr/lib64/mysql"],
       [MYSQL_LIBDIR="/usr/lib64/mysql"],
       [AS_IF([test -d "/usr/lib/mysql"],
          [MYSQL_LIBDIR="/usr/lib/mysql"],
          [MYSQL_LIBDIR="/usr/lib"])])
-    LDFLAGS="-L$MYSQL_LIBDIR $LDFLAGS $ZLIBS"
+    LDFLAGS="$LDFLAGS $ZLIBS"
     AC_CHECK_LIB([mysqlclient], [mysql_init],
       [AC_CHECK_HEADERS([mysql/mysql.h],
-         [MYSQL_LDFLAGS="-L$MYSQL_LIBDIR"
-          mysql=true],
+         [mysql=true],
          [mysql=false])],
       [mysql=false])])
 AC_SUBST(MYSQL_LDFLAGS)
@@ -1020,8 +1016,8 @@ AC_ARG_WITH([gnutls],
         [AC_CHECK_LIB([gnutls-dane], [dane_verify_crt_raw],
                       [gnutls_dane=1], [gnutls_dane=0])],
         [gnutls_dane=0])],
-     [LDFLAGS="-L$with_gnutls/lib $LDFLAGS"
-      CPPFLAGS="-I$with_gnutls/include $CPPFLAGS"
+     [LDFLAGS="$LDFLAGS"
+      CPPFLAGS="$CPPFLAGS"
       AC_CHECK_HEADERS([gnutls/abstract.h],
         [AC_CHECK_LIB([gnutls], [gnutls_priority_set],
                       [gnutls=1], [gnutls=0])],
